bed/modify.go

```diff
@@ -70,3 +71,176 @@ func MergeHighMem(records []Bed, mergeAdjacent bool) []Bed {
 	outList = append(outList, currentMax)
 	return outList
 }
+
+func mergeKeepLowNameAndScore(records []Bed) (out []Bed, newHidden bool) {
+	var dist int
+	newHidden = false
+	var outList = make([]Bed, 0)
+	SortByCoord(records)
+	var curr = records[0]
+	for i := 1; i < len(records); i++ {
+		if Overlap(curr, records[i]) {
+			if records[i].Score < curr.Score {
+				curr = records[i]
+			}
+		} else if curr.Chrom == records[i].Chrom {
+			dist = records[i].ChromStart - curr.ChromEnd
+			if (curr.Score + dist) < records[i].Score { //if the distance to the record on the left plus its hidden score
+				// is less than the hidden score stored for the right record, the right record will be reassigned the
+				//hidden that is equal to the dist to the left record plus it's hidden value and the gene name stored on the left record
+				newHidden = true
+				records[i].Score = curr.Score + dist
+				records[i].Name = curr.Name
+			} else if (records[i].Score + dist) < curr.Score {
+				curr.Score = records[i].Score + dist
+				curr.Name = records[i].Name
+			}
+			outList = append(outList, curr)
+			curr = records[i]
+		} else {
+			outList = append(outList, curr)
+			curr = records[i]
+		}
+	}
+	outList = append(outList, curr)
+	return outList, newHidden
+}
+
+func removeRecordsOnEmptyChrom(records []Bed, genome map[string]chromInfo.ChromInfo) []Bed {
+	var out = make([]Bed, 0)
+	var exist bool
+
+	for i := range records {
+		_, exist = genome[records[i].Chrom]
+		if exist {
+			out = append(out, records[i])
+		}
+	}
+	return out
+}
+
+// FillSpaceNoHiddenValue accepts a bed and reference genome as a map[string]chromInfo.ChromInfo and returns a bed
+// that assigns each genomic position to the nearest feature in the input bed using the Name field in the input bed.
+// Absolute start positions of the original input bed entries are stored in the Score field of the output bed entries.
+func FillSpaceNoHiddenValue(records []Bed, genome map[string]chromInfo.ChromInfo) []Bed {
```

Comment:
This does not seem like a general function for modifying beds, but rather something more specific to great, so I don't think bed/modify.go is the correct place.  Probably the 3dgreat folder is where this should be